middle_end/flambda2/from_lambda/dissect_letrec.ml

```diff
@@ -247,7 +247,10 @@ let rec prepare_letrec (recursive_set : Ident.Set.t)
             (id, def) :: defs, Lambda.Lvar id :: args)
         args ([], [])
     in
-    let arg_layout = Typeopt.layout_union layout_field layout_block in
+    (* [arg_layout] is more general than any of the possible layouts of [args]:
+       The arguments to [Pmakeblock]/[Pmakearray] are value fields, and the
+       argument to [Pduprecord] is a block. *)
+    let arg_layout = Typeopt.layout_union layout_value_field layout_block in
```

Comment:
Is this change because the fields can now be non-values, or is it a bug fix?

Reply:
It's not actually a change: `layout_field` and `layout_value_field` are defined to the same thing. But I carved up `layout_field` into several more specific names (including `layout_value_field`, for fields that are known to be values) in light of the fact that fields can now be non-values.